web/html/src/manager/content-management/list-filters/filter.utils.js

```diff
@@ -24,6 +26,9 @@ export function mapFilterFormToRequest(filterForm: FilterFormType, projectLabel:
   } else if (filterForm.type === filtersEnum.enum.ERRATUM.key) {
     requestForm.criteriaKey = "advisory_name";
     requestForm.criteriaValue = filterForm.advisoryName;
+  } else if (filterForm.type === filtersEnum.enum.ERRATUM_BYDATE.key) {
+    requestForm.criteriaKey = "issue_date";
+    requestForm.criteriaValue = filterForm.issueDate ? moment(filterForm.issueDate).format("YYYY-MM-DDTHH:mm:ssZ"):  "";
```

Comment:
@lneves12 - this produce a lint error:

```
error: Cannot resolve name `moment` (flowtype-errors/show-errors) at manager/content-management/list-filters/filter.utils.js:31:56:
```

I saw in a different file a line:

```
declare var moment: any;
```

But it does not look like a  variable. Wondering :-)

Reply:
That’s because this variable is being inject as a global variable by the package susemanager-frontend-libs, the old way of doing it :) 

As we are not importing it anywhere or saying what it is, flow is complaining. We can either import its types using https://github.com/flow-typed/flow-typed or just declare it as any, but with the latest we will not have any type checking while using moment.

I will need to refactor the way we manage timezones here, it’s not correct. I will handle this :)

Reply:
@lneves12 Sounds good. Doing this in a separat PR might be sufficient, what do you think?

Reply:
To be honest I am completely puzzled with our current timepicker date utils. I don't know how to solve this without a gigantic workaround or adding "moment-timezone".

If the server is on a different timezone than the timezone of the browser neither of these will work:
```
// function we use in other places to format the date from timepicker (Functions.Utils.dateWithTimezone)
Functions.Formats.LocalDateTime(this.state.earliest)   

moment(filterForm.issueDate).format("YYYY-MM-DDTHH:mm:ssZ")
```

As we can see on the image of this PR https://github.com/SUSE/spacewalk/pull/7172 if we use the LocalDateTime we will save the dates as strings on the DB without timezone context. That will be a problem if the customer ever changes the timezone on their server because all the stored filter dates will now be different.

I am already in the middle of an implementation using "moment-timezone"

Reply:
I added a commit adding the flow types for moment and assigning them automatically to the global instance.

Reply:
@mcalmer @ncounter @hustodemon I added a workaround for the timezones. I am not proud of it, but without doing a huge refactor it's what I could come up with: https://github.com/uyuni-project/uyuni/pull/1211/commits/67744014cbf595a494118e7f64359ff63548d29f

A review or ideas are welcome...

Reply:
A part from [this comment](https://github.com/uyuni-project/uyuni/commit/67744014cbf595a494118e7f64359ff63548d29f#r34602790) I left just to keep it in mind for the future, I am fine with the workaround. I don't think there is an easier and quicker way at the moment.